.github/workflows/main_matrix.yml

```
@@ -209,8 +209,32 @@ jobs:
+          case "${{ needs.check.result }}" in
+            success | skipped) ;;
+            *)
+              echo "::error::One or more checks failed or were cancelled (${{ needs.check.result }})."
+              exit 1
+              ;;
+          esac
+
   gather-artifacts:
-    if: github.repository == 'meshtastic/firmware'
+    # Only run on Release (workflow_dispatch) and Nightly (schedule) runs, not on PRs or merge_group runs.
+    if: github.repository_owner == 'meshtastic' && contains(github.event_name, fromJSON('["workflow_dispatch", "schedule"]'))
```

actionlint (1.7.12)

[error] 1st argument of function call is not assignable. "string" cannot be assigned to "array<any>". called function type is "contains(array<any>, any) -> bool"

(expression)